### PR TITLE
Move to Github Actions / Golangci-Lint

### PR DESCRIPTION
**Pull request by mrueg**
* Move from travis to Github actions.
* Use golangci-lint for linting

### CODE REVIEW

**Comment by mrueg** [comment]
@xiang90 @ptabor would you have time to look into this?

File: db_test.go [hunk 1]
```diff
@@ -30,15 +30,15 @@ const pageHeaderSize = 16
 
 // meta represents a simplified version of a database meta page for testing.
 type meta struct {
-	magic    uint32
```
Comment (dolmen) [inline db_test.go:33]: This change is not related to the advertised subject of the PR.

File: db_test.go [hunk 2]
```diff
@@ -30,15 +30,15 @@ const pageHeaderSize = 16
...
+	_        uint32
 	version  uint32
 	_        uint32
 	_        uint32
 	_        [16]byte
 	_        uint64
 	pgid     uint64
 	_        uint64
-	checksum uint64
```
Comment (dolmen) [inline db_test.go:41]: This change is not related to the advertised subject of the PR.

**Review by dolmen — CHANGES_REQUESTED** [review]
The changes in db_test.go should be in a separate PR.

File: db_test.go (hunk 1 above)
Comment (mrueg) [reply db_test.go:33]: It's required to make golangci-lint pass.

**Review by serathius — APPROVED** [review]
